packages/blockchain-wallet-v4-frontend/src/data/components/buySell/sagas.ts

```diff
@@ -502,18 +502,23 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
     }
   }
 
-  const checkAuthUrl = function* (orderId) {
+  const checkOrderAuthUrl = function* (orderId) {
     const order: ReturnType<typeof api.getBSOrder> = yield call(api.getBSOrder, orderId)
     if (order.attributes?.authorisationUrl || order.state === 'FAILED') {
       return order
     }
+
     throw new Error(BS_ERROR.RETRYING_TO_GET_AUTH_URL)
   }
 
   const orderConfirmCheck = function* (orderId) {
     const order: ReturnType<typeof api.getBSOrder> = yield call(api.getBSOrder, orderId)
 
-    if (order.state === 'FINISHED' || order.state === 'FAILED' || order.state === 'CANCELED') {
+    if (order.paymentError || order.state === 'FAILED' || order.state === 'CANCELED') {
```

Comment:
I know you didnt add all this code but it would be great to have this hardcoded strings in an enum constant